docs/Contributing/Fleet-UI-Testing.md

```diff
@@ -0,0 +1,362 @@
+(__as of 31/08/2022__) we do not have the best seperation of concerns between our systems. As a
+result, the most reliable way to test our software is as a whole with e2e tests. While this is ok
+at some level, we still are missing a large chunk of important testing that would be better written
+and maintained at the integration/unit level. We'd like to utilise seperation of concerns more
+between our systems as this will allow us to test more in the integration and unit layers which are
+quicker to run and generally easier to work with.
+
+### Types of Tests
+
+We use a variety of testing to ensure that our software is working as intended. This includes:
+
+- End-to-end (e2e)
+- Integration
+- Unit
+- Static
+- Manual
+
+#### Testing Cheat Sheet
+
+| Systems | Type of Test | Who is the User | Example | Notes |
+| ------- | ------------ | --------------- | ------- | ----- |
+| Reusuable utilites | unit w/jest | devs | string util, url util | no/few dependencies. function argument based |
+| Reusable hooks | unit w/jest & react-hooks lib | devs | useToggleDisplayed Hook | no/few dependencies. function argument based |
+| Reusable UI components | unit w/jest & testing-library | devs | Radio, Button, Input components | no/few dependencies. props based |
+| App Widgets | Integration or e2e w/testing-library or cypress | end users | Create User form, Reset Password form | Less reusable code with more complex environment setup and dependencies. Depending on the case can be done with integration or e2e. When integration, mock at the backend level, dont mock other UI systems. When e2e, no mocking of other systems. |
+| User Journeys | e2e w/cypress | end users | filtering a host by software, creating a team as admin | Full business flows. No/minimal mocking of systems, except for common error states from network. |
+| N/A | Manual | N/A | // TODO | Manual testing can be used for all types of code. Examples would be for one offs or states that would require extreamly difficult testing setups. |
+
+#### Manual Testing
+
+There will always be a space for manual testing and we utilise it for testing states that do not
+occur very often in the application or are not worth the effort to set up to test for an  unlikely edge case.
+
+#### Static Analysis
+
+This includes typing and linting to quickly ensure proper typings of data flowing through the application and that we are following coding conventions and styling rules. This give us a first of defense against writing buggy code.
+
+#### Unit Testing
+
+We unit test smaller reusable components that have no/few  dependencies within them. These
+components are primarily parametric based and require no or minimal mocking to be tested
+effectively. They tend to be small building blocks of our application (e.g. reusable UI components,
+common utilities, reusable hooks. The end user in this case tends to be other developers so we want
```

Comment:
End paranthesis missing

Reply:
RW:
"With unit testing, the end user tends to be other developers so we ensure these components work as expected when used as building blocks."